compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/LayoutConfiguration.desktop.kt

```diff
@@ -47,12 +47,21 @@ private val GraphicsConfiguration.density: Density get() = Density(
 
 internal val GlobalLayoutDirection get() = Locale.getDefault().layoutDirection
 
+internal val Locale.layoutDirection: LayoutDirection
+    get() = ComponentOrientation.getOrientation(this).layoutDirection
+
 internal val Component.layoutDirection: LayoutDirection
-    get() = this.locale.layoutDirection
+    get() = this.componentOrientation.layoutDirection
```

Comment:
componentOrientation is UKNOWN when we create the component (I checked).

We should read it from `locale` if it is the UKNOWN

Reply:
> I think we should do what was intended in Swing

I agree, but we have 2 issues:
1. We at least should set in Window/Dialog Composable functions, as they are not Swing to an external user.
2. We break behavior compatibility for ComposeWindow

I will check if Swing sets orientation after adding it into hirerachy tomorrow.

Reply:
> 2. We break behavior compatibility for ComposeWindow

I think, if we are sure that it is correct, and Swing doesn't set it by default, we can write in release notes about this.

Reply:
Hmm, looks like I was wrong. Swing doesn't set the component orientation by itself and the value remains `UNKNOWN`.

We could set it to `ComponentOrientation.getOrientation(this.locale)` in the constructor if we want to preserve the previous behavior.

Reply:
A safer option will be to preserve backward compatibility. The difference in behavior will be noticible only on RTL systems and only when Compose supports it and Swing doesn't.

Reply:
> This code changes the state of the component as non-obvious side effect, and can cause issues.
> We should also override setLocale:

But this also changes the state of the component as a non-obvious side effect, and it's an even stronger side effect. My version does the smallest change necessary: the locale is translated to the component orientation in the one specific place - the place where the (previous) code used to translated the locale to layout direction.

Also, overriding `setLocale` like that can lead to a surprising component orientation, like this:
```
window = ComposeWindow()
println(window.componentOrientation)  // UNKNOWN
window.locale = RtlLocale
println(window.componentOrientation)  // RIGHT_TO_LEFT
window.locale = LtrLocale
println(window.componentOrientation)  // RIGHT_TO_LEFT
```

But to be honest, I think both ways are not very good - maybe it's best to break backwards compatibility here. The previous behavior (setting layout direction from the locale rather than from component orientation) is incorrect, so maybe it's best to just remove it, and note in the release notes.

Reply:
> But this also changes the state of the component as a non-obvious side effect

It shouldn't change `componentOrientation`, only the internal `layoutDirection` that is passed to the scene:
```
window = ComposeWindow()
println(window.componentOrientation)  // UNKNOWN
window.locale = RtlLocale
println(window.componentOrientation)  // UNKNOWN, but the compose scene uses RTL
window.locale = LtrLocale
println(window.componentOrientation)  // UNKNOWN, but the compose scene uses LTR
```

Reply:
Ok, that can work.

Reply:
@igordmn Done